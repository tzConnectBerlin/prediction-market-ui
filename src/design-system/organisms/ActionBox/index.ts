export { ActionBox, ActionBox as default } from './ActionBox';
export type { ActionBoxProps } from './ActionBox';
