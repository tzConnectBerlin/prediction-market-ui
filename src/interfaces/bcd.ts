export interface SimilarContract {
  added?: number;
  address: string;
  alias?: string;
  annotations?: string[];
  balance?: number;
  count?: number;
  delegate?: string;
  delegate_alias?: string;
  entrypoints?: string[];
  fail_strings?: string[];
  found_by?: string;
  hardcoded?: string[];
  hash: string;
  id?: string;
  language?: string;
  last_action?: string;
  level?: number;
  manager?: string;
  migrations_count?: number;
  network?: string;
  project_id?: string;
  removed?: number;
  slug?: string;
  subscription?: Subscription;
  tags?: string[];
  timestamp?: string;
  tokens?: TokenBalance[];
  total_subscribed?: number;
  total_withdrawn?: number;
  tx_count?: number;
  verification_source?: string;
  verified?: boolean;
}

export interface Subscription {
  address?: string;
  alias?: string;
  network?: string;
  sentry_dsn?: string;
  sentry_enabled?: boolean;
  subscribed_at?: string;
  watch_calls?: boolean;
  watch_deployments?: boolean;
  watch_errors?: boolean;
  watch_mempool?: boolean;
  watch_migrations?: boolean;
  watch_same?: boolean;
  watch_similar?: boolean;
}

export interface TokenBalance {
  balance?: number;
  contract?: string;
  decimals?: number;
  name?: string;
  symbol?: string;
  token_id?: number;
}

export interface SimilarContractResponse {
  contracts?: SimilarContract[];
  count?: number;
}
