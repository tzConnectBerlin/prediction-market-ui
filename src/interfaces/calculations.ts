export interface ClosePositionReturn {
  aToSwap: number;
  aLeft: number;
  bReceived: number;
  aToSwapWithSlippage: number;
}

export interface ClosePositionBothReturn
  extends Omit<ClosePositionReturn, 'bReceived' | 'aToSwapWithSlippage'> {
  bHeld: number;
}

export interface BuyPosition {
  price: number;
  swap: number;
  quantity: number;
}
