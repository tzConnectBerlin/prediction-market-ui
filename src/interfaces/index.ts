export * from './wallet';
export * from './router';
