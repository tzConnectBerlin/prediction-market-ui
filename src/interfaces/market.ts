import { MichelsonMap } from '@taquito/taquito';
import { BigNumber } from 'bignumber.js';

export type QuestionType = string;

export interface Links {
  label: string;
  url: string;
}

export enum MarketStateType {
  auctionRunning = 'auction',
  marketBootstrapped = 'market',
}

export enum Role {
  participant = 'Participant',
  adjudicator = 'Adjudicator',
}

export interface AuctionNode {
  auctionRunningAuctionPeriodEnd: string;
  auctionRunningQuantity: string;
  auctionRunningYesPreference: string;
  auctionRunningUniswapContribution: string;
}

export interface MarketNode {
  auctionRewardCurrencyPool: string;
  liquidityRewardPool: string;
  marketCurrencyPool: string;
  bootstrapYesProbability: string;
  liquidityRewardSupplyUpdatedAtBlock: string;
  winningPrediction?: string;
  resolutionResolvedAtBlock?: number;
  marketBootstrappedBootstrappedAtBlock: string;
}

export interface PortfolioMarket {
  question: (string | undefined)[];
  holdings: string[] | string;
  price: string[] | string;
  total: string[] | string;
}

export interface PortfolioAuction {
  question: string;
  probability: string;
  quantity: string;
}

export interface MarketStateNode {
  node: AuctionNode | MarketNode;
}

export interface StorageMarketMapMarketBootstrappeds {
  nodes: MarketStateNode[];
}

export interface StorageMarketMapAuctionRunnings {
  nodes: MarketStateNode[];
}

export interface GraphMarketNode {
  node: GraphMarket;
}
export interface GraphMarket {
  id: number;
  block: number;
  deleted: boolean;
  marketId: string;
  metadataIpfsHash: string;
  metadataDescription: string;
  metadataAdjudicator: string;
  state: string;
  dateTime: {
    bakedAt: string;
  };
  storageMarketMapAuctionRunnings: StorageMarketMapAuctionRunnings;
  storageMarketMapMarketBootstrappeds: StorageMarketMapMarketBootstrappeds;
}

export interface IPFSMarketData {
  question: QuestionType;
  auctionEndDate: string;
  iconURL?: string;
  ticker: string;
}

export interface Market extends Partial<AuctionNode>, Partial<MarketNode>, IPFSMarketData {
  marketId: string;
  ipfsHash: string;
  description: string;
  adjudicator: string;
  state: MarketStateType;
  yesPrice: number;
  liquidity?: number | string;
  block: number;
  bakedAt: string;
}

export interface AllMarkets {
  markets: {
    marketNodes: GraphMarket[];
  };
}

export interface AllLedgers {
  ledgers: {
    ledgerMaps: LedgerMap[];
  };
}

export interface AllMarketsLedgers {
  markets: {
    marketNodes: GraphMarket[];
  };
  ledgers: {
    ledgerMaps: LedgerMap[];
  };
}

export interface MarketCardStatistic {
  type: string;
  value: string | number;
  changes?: string;
  currency?: Currency;
  tokenType?: TokenType;
}

export interface MarketCardToken {
  type: TokenType;
  value: number | string;
}

export interface MarketCardData extends Market {
  tokens?: MarketCardToken[];
  statistics?: MarketCardStatistic[];
}

export interface CreateMarket {
  marketId: number;
  ipfsHash: string;
  description: string;
  adjudicator: string;
  auctionEnd: string;
  initialBid: number;
  initialContribution: number;
  tokenType: 'fa12' | 'fa2';
  tokenAddress: string;
}

export interface GraphBet {
  probability: string;
  quantity: string;
}

export interface Bet {
  block: number;
  originator: string;
  probability: number;
  quantity: number;
  marketId: string;
}

export interface BetEdge {
  bet: GraphBet;
}

export interface GraphBets {
  totalBets: number;
  betEdges: BetEdge[];
}

export interface LqtProviderNode {
  id: number;
  marketId: string;
  originator: string;
  bets: GraphBets;
  block: number;
}

export interface LqtProviderEdge {
  lqtProviderNode: LqtProviderNode;
}

export interface StorageLiquidityProviderMaps {
  lqtProviderEdge: LqtProviderEdge[];
}

export interface AllBets {
  storageLiquidityProviderMaps: StorageLiquidityProviderMaps;
}

export enum MarketTradeType {
  payIn = 'payIn',
  payOut = 'payOut',
}

export enum FormType {
  buy = 'Buy',
  sell = 'Sell',
  addLiquidity = 'Add Liquidity',
  removeLiquidity = 'Remove Liquidity',
}

export interface LiquidityValues {
  probability: number;
  quantity: number;
}

export interface TokenSupplyMap {
  id: number;
  tokenId: string;
  totalSupply: string;
  tokenReserve: string;
  deleted: boolean;
}

export interface StorageSupplyMaps {
  supplyMaps: TokenSupplyMap[];
}

export interface AllTokens {
  storageSupplyMaps: StorageSupplyMaps;
}

export enum TokenType {
  yes = 'Yes',
  no = 'No',
}

export interface LedgerMap {
  id: number;
  block: number;
  deleted: boolean;
  owner: string;
  tokenId: string;
  quantity: string;
}

export interface StorageLedgerMaps {
  ledgerMaps: LedgerMap[];
}

export interface Token {
  id: number;
  tokenId: string;
  quantity: string;
  block: number;
  dateTime: {
    bakedAt: string;
  };
}

export interface TokenQuantity {
  token: Token[];
}

export interface MarketPricePoint {
  yesPrice: number;
  block: number;
  bakedAt: string;
}

export interface AddressTokens {
  tokenQuantity: TokenQuantity;
}

export interface AuctionMarkets {
  [key: string]: Market[];
}

// TODO: clean the stuff below

export interface QuestionState {
  [key: number]: {
    [key in MarketStateType]: symbol;
  };
}

export enum Currency {
  USD = '$',
  EUR = 'â‚¬',
}

export type CurrencyTypes = keyof typeof Currency;

export enum MarketEntrypoint {
  Bid = 'bid',
  BuyYes = 'buyYes',
  BuyNo = 'buyNo',
  ClaimWinnings = 'claimWinnings',
  CloseAuction = 'closeAuction',
  CreateQuestion = 'createQuestion',
  WithdrawAuction = 'withdrawAuction',
}

export interface CreateQuestion {
  question: QuestionType;
  auctionEndDate: Date | string;
  marketCloseDate: Date | string;
  iconURL?: string;
  yesAnswer: string;
  rate?: number;
  quantity?: number;
}

export interface Bid {
  question: QuestionType;
  rate: number;
  quantity: number;
}

export interface BuyToken {
  tokenType: TokenType;
  question: QuestionType;
  quantity: number;
}

export interface CloseMarket {
  tokenType: TokenType;
  question: QuestionType;
}

export interface ClaimWinnings {
  question: QuestionType;
}

export interface QuestionMetaData extends CreateQuestion {
  hash: string;
}

export interface BidEntry {
  rate: number;
  quantity: number;
  total_token: number;
}

export type BidRegistry = MichelsonMap<string, BidEntry>;

export interface QuestionEntry {
  owner: string;
  state: QuestionState;
  auction_end: string;
  market_close: string;
  auction_bids: BidRegistry;
  uniswap_pool?: unknown;
  tokens: {
    yes_token_id: BigNumber;
    no_token_id: BigNumber;
    lqt_token_id: BigNumber;
  };
  answer?: string | null;
  winning_token?: BigNumber | null; // token that will receive the reward
  total_auction_quantity: BigNumber;
  yes_preference: BigNumber;
  uniswap_contribution_factor: BigNumber;
}

export interface AuctionData {
  yes: number;
  no: number;
  participants: number;
}

export interface AuctionDataMap {
  [key: string]: AuctionData;
}

export interface MarketFilter {
  auctions: boolean;
  allMarkets: boolean;
  openMarkets: boolean;
  closedMarkets: boolean;
  onlyMyMarkets: boolean;
}

/**
 * Temp middleware interfaces
 */

export interface BidEntryMDW {
  rate: string;
  quantity: string;
  total_token: string;
}

export interface BidRegistryMDW {
  [key: string]: BidEntryMDW;
}

export interface QuestionEntryMDW {
  owner: string;
  state: {
    [key in MarketStateType]?: null;
  };
  auction_end: string;
  market_close: string;
  auction_bids: BidRegistryMDW;
  uniswap_pool?: unknown;
  tokens: {
    yes_token_id: string;
    no_token_id: string;
    lqt_token_id: string;
  };
  answer?: string | null;
  winning_token?: string | null;
  total_auction_quantity: string;
  yes_preference: string;
  uniswap_contribution_factor: string;
  price_yes?: string;
}

export interface QuestionEntryMDWMap {
  [key: string]: QuestionEntryMDW;
}

export interface StableCoinResponse {
  [key: string]: string;
}

export interface LedgerBalanceResponse {
  [tokenId: string]: { [address: string]: string };
}

export interface ContractError {
  [key: number]: string;
}

export interface DropDownItems {
  label: string;
  value: string | number;
}
