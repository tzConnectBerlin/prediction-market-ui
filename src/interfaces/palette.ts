import '@mui/material/styles';

declare module '@mui/material/styles' {
  interface Palette {
    tertiary: Palette['primary'];
    buttonText: Palette['text'];
  }
  interface PaletteOptions {
    tertiary: PaletteOptions['primary'];
    buttonText: PaletteOptions['text'];
  }
}
