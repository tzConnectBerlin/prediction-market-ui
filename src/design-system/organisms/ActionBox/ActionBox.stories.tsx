import * as React from 'react';
import { ToastProvider } from 'react-toast-notifications';
import { WalletProvider } from '@tezos-contrib/react-wallet-provider';
import { Story, Meta } from '@storybook/react/types-6-0';
import { ActionBox, ActionBoxProps } from './ActionBox';
import { MarketStateType } from '../../../interfaces';

export default {
  title: 'Organism/ActionBox',
  component: ActionBox,
} as Meta;

const Template: Story<ActionBoxProps> = (args) => (
  <WalletProvider name="Temple">
    <ToastProvider>
      <ActionBox {...args} />
    </ToastProvider>
  </WalletProvider>
);

export const Default = Template.bind({});

export const Resolved = Template.bind({});
Resolved.args = {
  marketId: '1',
  marketPhase: MarketStateType.marketBootstrapped,
  winningPrediction: 'yes',
};
