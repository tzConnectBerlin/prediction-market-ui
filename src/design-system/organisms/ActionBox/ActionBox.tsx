import * as React from 'react';
import { Card, CardContent, useTheme } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { MarketStateType } from '../../../interfaces';
import Button from '../../atoms/Button';
import { Typography } from '../../atoms/Typography';
import { ResolveMarketModal } from '../ResolveMarketModal';
import { ResolveMarketForm } from '../ResolveMarketModal/ResolveMarketModal';

export interface ActionBoxProps {
  marketId: string;
  marketPhase: MarketStateType;
  adjudicator?: string;
  address?: string;
  winningPrediction?: string;
  handleCloseAuction?: (id: string) => Promise<void>;
  handleResolveMarket?: (values: ResolveMarketForm) => Promise<void>;
  closeMarketId?: string;
  setCloseMarketId?: React.Dispatch<React.SetStateAction<string>>;
  auctionParticipant?: boolean;
}

const StyledCard = styled(Card)`
  margin-bottom: 1.5rem;
  padding: 1rem;
  padding-bottom: 0.5rem;
`;

const StyledDiv = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;
export const ActionBox: React.FC<ActionBoxProps> = ({
  adjudicator,
  address,
  winningPrediction,
  marketPhase,
  marketId,
  handleCloseAuction,
  handleResolveMarket,
  closeMarketId,
  setCloseMarketId,
  auctionParticipant,
}) => {
  const theme = useTheme();
  const handleClose = () => setCloseMarketId && setCloseMarketId('');
  const { t } = useTranslation('common');
  const winningPredictionText =
    winningPrediction && winningPrediction.charAt(0).toUpperCase() + winningPrediction.slice(1);

  return marketPhase === MarketStateType.marketBootstrapped &&
    adjudicator !== address &&
    !winningPrediction ? null : (
    <StyledCard>
      {handleResolveMarket && (
        <ResolveMarketModal
          open={!!closeMarketId}
          handleClose={handleClose}
          handleSubmit={handleResolveMarket}
        />
      )}
      <CardContent>
        {marketPhase === 'auction' && (
          <Typography marginBottom="1.5rem" size="0.875rem">
            {t('closeAuction')}
          </Typography>
        )}
        {winningPrediction && (
          <StyledDiv>
            <Typography>{t('resolution')}</Typography>
            <Typography
              color={
                winningPrediction === 'yes' ? theme.palette.success.main : theme.palette.error.main
              }
            >
              {winningPredictionText}
            </Typography>
          </StyledDiv>
        )}
        {((!winningPrediction && adjudicator === address) ||
          (marketPhase === 'auction' && auctionParticipant)) && (
          <Button
            lowercase
            fullWidth
            label={marketPhase === 'auction' ? t('openMarketToTrade') : t('closeMarket')}
            onClick={
              marketPhase === 'auction'
                ? () => handleCloseAuction && handleCloseAuction(marketId)
                : () => setCloseMarketId && setCloseMarketId(marketId)
            }
          />
        )}
      </CardContent>
    </StyledCard>
  );
};
