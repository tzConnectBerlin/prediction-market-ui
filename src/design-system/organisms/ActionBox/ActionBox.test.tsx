import renderer from 'react-test-renderer';
import { fireEvent, getAllByText, render } from '@testing-library/react';
import { ActionBox, ActionBoxProps } from './ActionBox';
import { MarketStateType } from '../../../interfaces';

const closeMock = jest.fn();
const resolveMock = jest.fn();
const setMock = jest.fn();

const defaultArgs: ActionBoxProps = {
  adjudicator: 'tim',
  marketId: '1',
  marketPhase: MarketStateType.auctionRunning,
  handleCloseAuction: closeMock,
  handleResolveMarket: resolveMock,
  auctionParticipant: true,
  setCloseMarketId: setMock,
  closeMarketId: '',
};

const WrappedComponent: React.FC<any> = (args: any) => <ActionBox {...args} />;

describe('Snapshot - render TradeForm', () => {
  it('renders correctly with default props', () => {
    const BuyForm = renderer.create(<WrappedComponent {...defaultArgs} />).toJSON();
    expect(BuyForm).toMatchSnapshot();
  });
  it('calls close function', () => {
    const { getByRole } = render(<WrappedComponent {...defaultArgs} />);
    fireEvent.click(getByRole('button'));
    expect(closeMock).toBeCalled();
  });
  it('calls set function', () => {
    const { getByRole } = render(
      <WrappedComponent
        {...defaultArgs}
        marketPhase={MarketStateType.marketBootstrapped}
        address="tim"
      />,
    );
    fireEvent.click(getByRole('button'));
    expect(setMock).toBeCalled();
  });
  it('has a winning prediction', () => {
    const { findByRole, getByText } = render(
      <WrappedComponent
        {...defaultArgs}
        marketPhase={MarketStateType.marketBootstrapped}
        winningPrediction="yes"
      />,
    );
    expect(findByRole('button')).toMatchObject({});
    expect(getByText(/yes/i)).toBeInTheDocument();
  });
});
