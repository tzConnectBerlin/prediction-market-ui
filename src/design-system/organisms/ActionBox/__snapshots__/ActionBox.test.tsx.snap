// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot - render TradeForm renders correctly with default props 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  margin-bottom: 1.5rem;
  padding: 1rem;
  padding-bottom: 0.5rem;
}

.emotion-1 {
  padding: 16px;
}

.emotion-1:last-child {
  padding-bottom: 24px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  font-size: 0.875rem;
  margin-bottom: 1.5rem;
  font-size: 0.875rem;
}

.emotion-2.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 4px 10px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
  border-radius: 0.2em;
  padding: 0.2em 1.2em;
  border: solid 2px transparent;
  text-transform: none;
  box-shadow: none;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: #1976d2;
  }
}

.emotion-3:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-3.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-3:hover {
  border-width: 2px;
  box-shadow: none;
}

.emotion-3:disabled {
  border-color: transparent;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
>
  <div
    className="MuiCardContent-root emotion-1"
  >
    <p
      className="MuiTypography-root MuiTypography-body1  emotion-2"
      lines={3}
    >
      closeAuction
    </p>
    <button
      borderColor="transparent"
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root emotion-3"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      textType="none"
      type="button"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3  emotion-4"
        lines={3}
      >
        openMarketToTrade
      </h3>
    </button>
  </div>
</div>
`;
